meetings/working-groups/collection-literals/CL-2023-08-03.md

```diff
@@ -0,0 +1,146 @@
+
+Similar to option 1, it felt very strange for us to introduce behavior where the compiler would heap allocate for the user, only to warn since in the majority of times it was believed users would not want that.  After all, in the case of heap allocating, one might as well just choose a heap allocated type *to begin with*.
+
+These concerns led us to our final option:
+
+### Option 3. Always stack-alloc for ref-struct target type.
+
+We then moved to a discussion around our belief that when using ref-structs, and collection expressions, users would expect that data to be put on the stack.  Exploring this led to the following semantics:
+
+```c#
+ReadOnlySpan<int> x1 = [a, b, c];        // stack allocated, local scope.
+scoped ReadonlySpan<int> x2 = [a, b, c]; // stack allocated, local scope.
+```
+
+In other words, the above two have the same meaning.  With the latter just being explicit about the local scoping.  Returning either of those from a method would be illegal.  This has the benefit that the simple form matches the intuition nearly all users will want this to have, and will provide the best performance by default.  This heavily aligns with the stated goal of collection expressions (which we will continue to evangelize it with) that it makes the right default choices and by using it you should always (*ideally*) be getting the best performance results.  
+
+This does have some downsides though.
+
+First, if the user did want to return the ref struct, they would now be blocked.  The workaround for this would be to instead be explicit that you wanted heap allocation for this span (e.g. `ReadOnlySpan<int> x1 = (int[])[1, 2, 3];`).  However, we see this explicitness as a *good* thing.  When working with spans and collection-expressions, we feel that having code be explicit when it is on the heap is actually a desirable documentary property for it to have.  And, while the cast syntax is not attractive, we could consider a simpler way of expressing that in the future (e.g. `new [1, 2, 3]`), where the syntax was light, but clear about intent.
+
+Second, the aforementioned redundancy if you actually include `scoped`.  We could consider warning or blocking that to prevent confusion and to help convey that the data is always scoped.  That said, we already have this redundancy with `scoped ReadONlySpan<int> x = stackalloc int[] { ... };` and we give no warning.  So we likely can just do the same here.
```

Comment:
```suggestion
Second, the aforementioned redundancy if you actually include `scoped`.  We could consider warning or blocking that to prevent confusion and to help convey that the data is always scoped.  That said, we already have this redundancy with `scoped ReadOnlySpan<int> x = stackalloc int[] { ... };` and we give no warning.  So we likely can just do the same here.
```